src/components/grouped-character/grounded-character.test.tsx

```diff
@@ -0,0 +1,604 @@
+          onBlur?.(ev.target.value);
+        }}
+      />
+    );
+  };
+
+  render(<MockComponent />);
+
+  input = screen.getByRole("textbox");
+
+  await user.click(input);
+  await user.keyboard("{arrowLeft}");
+  await user.keyboard("{arrowRight}");
+  await user.keyboard("{arrowRight}");
+  await user.keyboard("{arrowRight}");
+  await user.keyboard("{arrowRight}");
+  jest.runAllTimers();
+
+  expect(input.selectionStart).toBe(4);
+  expect(input.selectionEnd).toBe(4);
+
+  await user.keyboard("{backspace}");
+  jest.runAllTimers();
+
+  expect(input.selectionStart).toBe(2);
+  expect(input.selectionEnd).toBe(2);
+});
+
+test("adding a character before a separator moves cursor forwards two positions", async () => {
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+  const MockComponent = () => {
+    const [stateValue, setStateValue] = useState<string>("123123");
+    return (
+      <GroupedCharacter
+        separator="-"
+        value={stateValue}
+        groups={[2, 2, 4]}
+        onChange={(ev) => {
+          setStateValue(ev.target.value.rawValue);
+          onChange?.(ev.target.value);
+        }}
+        onBlur={(ev) => {
+          setStateValue(ev.target.value.formattedValue);
+          onBlur?.(ev.target.value);
+        }}
+      />
+    );
+  };
+
+  render(<MockComponent />);
+
+  input = screen.getByRole("textbox");
+
+  await user.click(input);
+  await user.keyboard("{arrowLeft}");
+  await user.keyboard("{arrowRight}");
+  await user.keyboard("{arrowRight}");
+  await user.keyboard("{arrowRight}");
+  await user.keyboard("{arrowRight}");
+  await user.keyboard("{arrowRight}");
+  jest.runAllTimers();
+
+  expect(input.selectionStart).toBe(5);
+  expect(input.selectionEnd).toBe(5);
+
+  await user.keyboard("1");
+  jest.runAllTimers();
+
+  expect(input.selectionStart).toBe(7);
+  expect(input.selectionEnd).toBe(7);
+});
+
+test("when no character is present, cursor position goes to 0", async () => {
+  const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+  const MockComponent = () => {
+    const [stateValue, setStateValue] = useState<string>("");
+    return (
+      <GroupedCharacter
+        separator="-"
+        value={stateValue}
+        groups={[2, 2, 4]}
+        onChange={(ev) => {
+          setStateValue(ev.target.value.rawValue);
+          onChange?.(ev.target.value);
+        }}
+        onBlur={(ev) => {
+          setStateValue(ev.target.value.formattedValue);
+          onBlur?.(ev.target.value);
+        }}
+      />
+    );
+  };
+
+  render(<MockComponent />);
+
+  input = screen.getByRole("textbox");
+
+  await user.click(input);
+  jest.runAllTimers();
+
+  expect(input.selectionStart).toBe(0);
+  expect(input.selectionEnd).toBe(0);
+});
+
+test("pressing backspace when the cursor is after the middle character in the final group, should move the cursor one position backwards", async () => {
```

Comment:
**comment** (non-blocking): this test description is quite confusing, mostly because it seems weirdly specific. Is it that important that it's specifically the middle character in the final group (that only makes sense if the final group is of size 3, or I guess 5,7,etc)? I suspect this is really just testing that if you press backspace and you're not just after a separator, then it only goes back 1 space - but I'm far from certain. If that is the case I would try to simplify the test description so that it isn't mentioning a very specific case which isn't relevant.

Reply:
Yeah it's to hit a `else if` condition in the `lastPosition` block in the `handleChange` function. I might change it to include the words that roughly mean `isAtOneBeyondSeparator` as that is where this test covers in the code.

Code block I am referring to: 
https://github.com/Sage/carbon/blob/master/src/components/grouped-character/grouped-character.component.tsx#L109-L112

Reply:
thanks, the description reads much better now 👍